libhoney/transmission.py

```diff
@@ -84,19 +84,33 @@ def _send(self, ev):
             "X-Honeycomb-Team": ev.writekey,
             "X-Honeycomb-SampleRate": str(ev.sample_rate)})
         preq = self.session.prepare_request(req)
-        resp = self.session.send(preq)
-        if (resp.status_code == 200):
-            sd.incr("messages_sent")
-        else:
+        try:
+            resp = self.session.send(preq)
+            if (resp.status_code == 200):
+                sd.incr("messages_sent")
+            else:
+                sd.incr("send_errors")
+            dur = get_now() - start
+            response = {
+                "status_code": resp.status_code,
+                "duration": dur.total_seconds() * 1000,  # report in milliseconds
+                "metadata": ev.metadata,
+                "body": resp.text,
+                "error": "",
+            }
+        except requests.exceptions.ConnectionError as e:
```

Comment:
Any reason to not expand the `try`/`except` to cover the entire method and remove the limitation to `ConnectionError` here?